windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -0,0 +1,535 @@
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+using ch.cyberduck.core.i18n;
+using ch.cyberduck.core.local;
+using ch.cyberduck.core.preferences;
+using ch.cyberduck.core.transfer;
+using ch.cyberduck.ui.Model;
+using Ch.Cyberduck.Core.Refresh.Services;
+using Ch.Cyberduck.Ui.Controller;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DynamicData;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Windows.Media;
+using static Ch.Cyberduck.ImageHelper;
+
+namespace ch.cyberduck.ui.ViewModels;
+
+public partial class TransfersViewModel : SynchronizedObservableObject
+{
+    private readonly Dictionary<int, BandwidthViewModel> bandwidthLookup;
+    private readonly TransferController controller;
+    private readonly WpfIconProvider iconProvider;
+    private readonly Preferences preferences;
+    private readonly ReadOnlyObservableCollection<TransferViewModel> selectedTransfers;
+    private readonly TransfersStore store;
+    private readonly ReadOnlyObservableCollection<TransferViewModel> transfers;
+    private readonly Dictionary<int, ConnectionViewModel> connectionLookup;
+    private BandwidthViewModel selectedBandwidth;
+    [ObservableProperty]
+    private ConnectionViewModel selectedConnectionLimit;
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(BandwidthEnabled))]
+    [NotifyCanExecuteChangedFor(nameof(OpenCommand))]
+    [NotifyCanExecuteChangedFor(nameof(ReloadCommand))]
+    [NotifyCanExecuteChangedFor(nameof(RemoveCommand))]
+    [NotifyCanExecuteChangedFor(nameof(ResumeCommand))]
+    [NotifyCanExecuteChangedFor(nameof(ShowCommand))]
+    [NotifyCanExecuteChangedFor(nameof(StopCommand))]
+    [NotifyCanExecuteChangedFor(nameof(TrashCommand))]
+    private TransferViewModel selectedTransfer;
+    [ObservableProperty]
+    private ImageSource selectedTransferFileIcon;
+    [ObservableProperty]
+    private string selectedTransferLocal;
+    [ObservableProperty]
+    private string selectedTransferUrl;
+
+    public ObservableCollection<BandwidthViewModel> Bandwidth { get; }
+
+    public bool BandwidthEnabled => SelectedTransfer is not null;
+
+    public ObservableCollection<ConnectionViewModel> Connections { get; }
+
+    public BandwidthViewModel SelectedBandwidth
+    {
+        get => selectedBandwidth;
+        set
+        {
+            if (SetProperty(ref selectedBandwidth, value))
+            {
+                OnSelectedBandwidthChanged(value);
+            }
+        }
+    }
+
+    public ReadOnlyObservableCollection<TransferViewModel> SelectedTransfers => selectedTransfers;
+
+    public bool ToolbarCleanup
```

Comment:
Maybe move away from here, as it doesn't belong here, but as the preferences aren't reactive/observable/INPC-capable, properties are stored here for the time being.